src/redux/actions.js

```diff
@@ -0,0 +1,41 @@
+import axios from 'axios';
+import {
+	LOADING_REPO,
+	LOADING_USER,
+	SEARCH_REPO_FAILURE,
+	SEARCH_REPO_SUCCESS,
+	SEARCH_USER_FAILURE,
+	SEARCH_USER_SUCCESS,
+} from './constants';
+
+/**
+ * Get information about a user
+ * @param {string} username used to search for a specific user's information
+ */
+export const getUserInfo = (username) => async (dispatch) => {
+	dispatch({ type: LOADING_USER });
+
+	try {
+		const { data } = await axios.get(`https://api.github.com/users/${username}`);
+		dispatch({ type: SEARCH_USER_SUCCESS, payload: data });
+	} catch (error) {
+		dispatch({ type: SEARCH_USER_FAILURE, payload: error.message });
```

Comment:
im not sure this works, because I tried putting a rubbish fake username in and it just refreshed the search page back to an empty search box. when on a persons profile page, searching a fake user also does nothing. something to look into?

Reply:
will fix this

Reply:
toby will